From 32e5c1821352e034c946356576b62811d95e26e7 Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Tue, 9 May 2023 15:39:10 -0700
Subject: [PATCH] [quartz] Pad both the content rect and the window width.

Otherwise the rastering gets messed up and popup windows get badly
truncated lengthwise on non-Retina monitors when there are both
Retina and non-Retina monitors.
---
 gdk/quartz/GdkQuartzNSWindow.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gdk/quartz/GdkQuartzNSWindow.c b/gdk/quartz/GdkQuartzNSWindow.c
index 5a546891aa..6a3d2919c1 100644
--- a/gdk/quartz/GdkQuartzNSWindow.c
+++ b/gdk/quartz/GdkQuartzNSWindow.c
@@ -245,7 +245,10 @@
   window->height = content_rect.size.height;
 
   if(window->width % align)
-    content_rect.size.width += align - window->width % align;
+    {
+      window->width += align - window->width % align;
+      content_rect.size.width = window->width;
+    }
 
   content_rect.origin.x = 0;
   content_rect.origin.y = 0;
-- 
2.37.1 (Apple Git-137.1)

